lib/manageiq/api/client/authentication.rb

```diff
@@ -0,0 +1,38 @@
+module ManageIQ
+  module API
+    module Client
+      class Authentication
+        attr_accessor :user, :password, :token, :group
+
+        DEFAULTS = {
+          :user     => "admin",
+          :password => "smartvm"
+        }.freeze
+
+        def initialize(options = {})
+          @user, @password = fetch_credentials(options)
+          @token = options[:token]
+          @group = options[:group]
+
+          unless token
+            raise "Must specify both a user and a password" if user.blank? || password.blank?
+          end
+        end
+
+        def inspect
+          super.gsub(/@password=\".+?\", /, "")
```

Comment:
If the password is at the end of the attributes list it won't be removed because of the tailing `, `.  So, if someone changes the order of the attr_accessor above this will probably fail.  A test would just pick that up, but it might be better to just code for the tailing `, ` as optional.